### ISSUE TITLE
Can't use react-iconography by itself.

### ISSUE THREAD

**Issue by kud** [pivotal-cf/pivotal-ui#423]
```
Error in ./~/pui-react-iconography/iconography.js
Module not found: 'svg-react' in /project/node_modules/pui-react-iconography

 @ ./~/pui-react-iconography/iconography.js 55:13-80
```

I've got that if I use iconography itself. And if I install `svg-react-loader`, i've got no error but lots of warning.

**Comment by charleshansen** [comment]
The problem may be very weird timing, are you using version 7?

 I'm doing a soft launch of version 7 to make sure that at least a couple teams can upgrade without major bugs. The documentation on the styleguide is for version 6, and iconography changed significantly in that time. If you want a sneak peak at the version 7 documentation, you can look at styleguide-staging.cfapps.io (this is not normally recommended, but works now). Hopefully I'll push up the version 7 styleguide soon.

**Comment by kud** [comment]
I saw you changed some things with the new version.

I tried with "src" and it works better yes. But the thing is I use component by component, not the whole ui framework.

**Comment by kud** [comment]
And you could/should add an "import" version too instead of "require" I think. :)

**Comment by kud** [comment]
I will have a try with http://styleguide-staging.cfapps.io/react_base_iconography.html

I use create-react-app to be fair.

**Comment by kud** [comment]
This is what I've got when I install and use all what you said on the doc:

```html
<span class="svgicon"><svg class="icon-add" style="width: 100px; height: 100px;"></svg></span>
```

but nothing is displayed.

Plus:

```
Warning in ./~/pui-react-iconography/~/pui-css-iconography/svgs/warning.svg
Module build failed: Error: Non-whitespace before first tag.
Line: 0
Column: 1
Char: m
 @ ./~/pui-react-iconography/~/pui-css-iconography/svgs ./~/react-scripts/~/babel-loader!./~/svg-react-loader!^\.\/.*\.svg$

Warning in ./~/pui-react-iconography/~/pui-css-iconography/svgs/zoom_in.svg
Module build failed: Error: Non-whitespace before first tag.
Line: 0
Column: 1
Char: m
 @ ./~/pui-react-iconography/~/pui-css-iconography/svgs ./~/react-scripts/~/babel-loader!./~/svg-react-loader!^\.\/.*\.svg$

Warning in ./~/pui-react-iconography/~/pui-css-iconography/svgs/zoom_out.svg
Module build failed: Error: Non-whitespace before first tag.
Line: 0
Column: 1
Char: m
 @ ./~/pui-react-iconography/~/pui-css-iconography/svgs ./~/react-scripts/~/babel-loader!./~/svg-react-loader!^\.\/.*\.svg$
```

etc etc.

**Comment by mattrothenberg** [comment]
@kud When you inspect the SVG element in Chrome, what do you see? Could you add a screenshot?

**Comment by kud** [comment]
For the moment, I've removed this package. And I was on firefox nightly. I'll let you know when i'll test it.

**Comment by charleshansen** [comment]
The 'Non-whitespace before first tag' error is because you have secret extra svg loaders that are converting the svg before it gets to the svg-react-loader.

Looking through create-react-app, they convert any svg into a file to be exported as an asset. This is normally great if you aren't doing anything fancy, but it will break svg-react-loader. I'm not sure how to turn this off in create-react-app. I suggest that you don't use create-react-app. It is great for getting started, but really hard to customize in any way. We provide our own [starter app](https://github.com/pivotal-cf/react-starter) if you want to give it a try. There are a million react starters out there, just pick one you can configure.

**Comment by kud** [comment]
It's for a PoC so create-react-app is amazing. If not, i'll do my own process, yes. ;)

**Comment by charleshansen** [comment]
Yeah, create-react-app is pretty amazing out of the box. It clearly took a lot of time to build it.

For what it is worth, you can still use the Pivotal UI SVGs, you just can't use them with React, and it is more annoying. If you still need PUI icons, check out the documentation for the CSS Iconography component (the staging styleguide still).

**Comment by kud** [comment]
Thank you @charleshansen for your time. Yes I'll do it this way I think.

**Comment by robcaldecott** [comment]
We are using create-react-app and cannot use some of the Pivotal components because of this Webpack SVG dependency. The Dropdown component uses SVG icons internally for example. We could eject our CRA projects and manually edit the Webpack config but this means we lose out on future CRA updates so this would be a last resort. Is there no way to avoid this SVG dependency?

**Comment by charleshansen** [comment]
Sorry, I haven't really dealt with this yet. There may be a solution, but I'm not sure what it is. I do still think that you should not use CRA for a real website though. Pivotal UI is just the first of many issues you will run into if you can't control your webpack config.

**Comment by charleshansen** [comment]
I just published a fix in 7.0.2 to address this issue. The SVGs should compile correctly now, ignoring any webpack config your project is injecting in front of the svg loader.

Create React App has the separate problem that you need to get the css for the components. pui-react-tools extracts the css for you using ExtractTextPlugin. You can always just download all of the css at styleguide.pivotal.io/components.css

**Comment by robcaldecott** [comment]
Fantastic, can't wait to try it. We used that Dr Frankenstyle tool to extract the styles but am happy to use the full CSS as I think you're deprecating that tool.
